### DISCUSSION TITLE
How can I export a notebook to HTML format?

### DISCUSSION THREAD

**Discussion by derekmahar** [xwmx/nb#262] [Q&A]
How can I export every Markdown note in a notebook to HTML format where [nb](https://xwmx.github.io/nb/) converts [`[[wiki-style links]]`](https://xwmx.github.io/nb/#-linking) to HTML hyperlinks?

**Comment by xwmx** [comment]
@derekmahar I think that might not be easily possible yet. You could in theory iterate over each file and then print the HTML for the `nb` item page (including CSS and extra markup) with `nb show <id> --render --print --no-color`. I'll have to think about this a bit.

**Reply by derekmahar** [reply]
How does `nb browse` render the wiki-style links?  I find that on my Raspberry Pi 4, following these links using `nb browse` is sluggish compared to following Web links.

**Reply by xwmx** [reply]
There is a [`_resolve_links()`](https://github.com/xwmx/nb/blob/master/nb#L11647) function that rewrites the `[[wiki-style links]]` to URLs and is used within rendering pipelines that exist in `_browse()` and `_show()`.

**Reply by derekmahar** [reply]
Do any of the steps in the rendering pipeline invoke Pandoc?

**Reply by xwmx** [reply]
Yes. Rendering pipelines going from Pandoc-supported input formats (Markdown, `.docx`, etc) to HTML use Pandoc.

**Reply by derekmahar** [reply]
Does the rendering pipeline cache any Pandoc output?  I'm guessing it does not because I've noticed the same delay when I repeatedly follow a wiki-style link.

**Reply by xwmx** [reply]
No caching.

**Reply by derekmahar** [reply]
I'm impressed that you've managed to implement so many features using only bash and other common command line tools.

**Reply by xwmx** [reply]
Thank you! 😊🙏

**Reply by derekmahar** [reply]
> @derekmahar I think that might not be easily possible yet. You could in theory iterate over each file and then print the HTML for the `nb` item page (including CSS and extra markup) with `nb show <id> --render --print --no-color`. I'll have to think about this a bit.

I noticed that the wiki-style links in the HTML that `nb show <id> --render --print --no-color` produces refer to the original target document (in this case, Markdown), rather than to the HTML document to which that target Markdown document would render.  To generate a working website, `nb show <id> --render --print --no-color` would have to rewrite all of the HTML links to point to the corresponding HTML documents.

Are you familiar with any command line tools that convert Markdown documents containing wiki-style links to HTML?

**Reply by derekmahar** [reply]
> Are you familiar with any command line tools that convert Markdown files containing wiki-style links to HTML?

I found some [solutions](https://stackoverflow.com/questions/40993488/convert-markdown-links-to-html-with-pandoc) which use Python or Lua Pandoc filters that convert links to Markdown documents to links to HTML documents, though these filters don't convert wiki-style links.  To convert wiki-style links, I would need an intermediate filter that converts links in the form `[[Title A]]` to Markdown links in the form `[Title A](title_a.md)`.
